Controller/Adminhtml/Menu/Edit.php

```diff
@@ -1,58 +1,80 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Snowdog\Menu\Controller\Adminhtml\Menu;
 
-use Magento\Backend\App\Action;
-use Magento\Framework\App\ResponseInterface;
-use Magento\Framework\Controller\ResultFactory;
-use Magento\Framework\Exception\NoSuchEntityException;
+use Snowdog\Menu\Controller\Adminhtml\MenuAction;
+use Magento\Framework\App\Action\HttpGetActionInterface;
+use Magento\Backend\App\Action\Context;
 use Magento\Framework\Registry;
+use Magento\Framework\View\Result\PageFactory;
 use Snowdog\Menu\Api\MenuRepositoryInterface;
+use Snowdog\Menu\Model\MenuFactory;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Backend\Model\View\Result\Redirect;
+use Magento\Backend\Model\View\Result\Page;
 
-class Edit extends Action
+class Edit extends MenuAction implements HttpGetActionInterface
 {
     const REGISTRY_CODE = 'snowmenu_menu';
+
     /**
-     * @var MenuRepositoryInterface
+     * @var PageFactory
      */
-    private $menuRepository;
+    private $resultPageFactory;
+
     /**
      * @var Registry
      */
-    private $registry;
+    private $coreRegistry;
 
-    public function __construct(Action\Context $context, MenuRepositoryInterface $menuRepository, Registry $registry)
-    {
-        parent::__construct($context);
-        $this->menuRepository = $menuRepository;
-        $this->registry = $registry;
+    /**
+     * @param Context $context
+     * @param Registry $coreRegistry
+     * @param PageFactory $resultPageFactory
+     * @param MenuRepositoryInterface $menuRepository
+     * @param MenuFactory $menuFactory
+     */
```

Comment:
I think we can remove `__construct` docs as all params types are already defined within the method declaration

Reply:
Done.